data_safe_haven/infrastructure/programs/sre/networking.py

```diff
@@ -1282,6 +1341,18 @@ def __init__(
                     source_address_prefix=SREIpRanges.workspaces.prefix,
                     source_port_range="*",
                 ),
+                network.SecurityRuleArgs(
+                    access=network.SecurityRuleAccess.ALLOW,
+                    description="Allow outbound connections to configuration data endpoints.",
+                    destination_address_prefix=SREIpRanges.data_configuration.prefix,
+                    destination_port_range="*",
+                    direction=network.SecurityRuleDirection.OUTBOUND,
+                    name="AllowDataConfigurationEndpointsOutbound",
+                    priority=NetworkingPriorities.INTERNAL_SRE_DATA_CONFIGURATION,
+                    protocol=network.SecurityRuleProtocol.ASTERISK,
+                    source_address_prefix=SREIpRanges.workspaces.prefix,
+                    source_port_range="*",
+                ),
```

Comment:
```suggestion
                network.SecurityRuleArgs(
                    access=network.SecurityRuleAccess.ALLOW,
                    description="Allow outbound connections to desired state data endpoints.",
                    destination_address_prefix=SREIpRanges.data_desired_state.prefix,
                    destination_port_range="*",
                    direction=network.SecurityRuleDirection.OUTBOUND,
                    name="AllowDataDesiredStateEndpointsOutbound",
                    priority=NetworkingPriorities.INTERNAL_SRE_DATA_DESIRED_STATE,
                    protocol=network.SecurityRuleProtocol.ASTERISK,
                    source_address_prefix=SREIpRanges.workspaces.prefix,
                    source_port_range="*",
                ),
```

N.B. Do we not know which destination port this connection will use?

Reply:
I knew I would miss something :smile:, quite a few places to make changes.

Reply:
> The NFS 3.0 protocol uses ports 111 and 2049. If you're connecting from an on-premises network, make sure that your client allows outgoing communication through those ports. If you have granted access to specific VNets, make sure that any network security groups associated with those VNets don't contain security rules that block incoming communication through those ports.

https://learn.microsoft.com/en-us/azure/storage/blobs/network-file-system-protocol-support#supported-network-connections

Reply:
Fixed in ed11e31 and 69674c2